backend/addcorpus/reader.py

```diff
@@ -0,0 +1,38 @@
+import glob
+
+from addcorpus.models import Corpus, Field
+from addcorpus.python_corpora.load_corpus import load_corpus_definition
+from ianalyzer_readers.extract import CSV
+from ianalyzer_readers.readers.core import Field as ReaderField
+from ianalyzer_readers.readers.core import Reader
+from ianalyzer_readers.readers.csv import CSVReader
+
+
+def make_reader_field(corpus_field: Field) -> ReaderField:
+    return ReaderField(
+        name=corpus_field.name,
+        extractor=CSV(corpus_field.extract_column),
+    )
+
+
+def make_reader(corpus: Corpus, data_directory: str = None) -> Reader:
+    '''
+    From a corpus, returns a Reader object that allows source extraction
+
+    For Python corpora, simply loads the definition class,
+    for JSON based corpora, construct Reader from the database.
+    '''
+    if corpus.has_python_definition:
+        return load_corpus_definition(corpus.name)
```

Comment:
It's a bit inconsistent that `data_directory` is required for corpora without a Python definition, but completely ignored without one. Especially since you could otherwise call `make_reader` without caring about whether the corpus has a Python definition or not.

Perhaps we can fill in the directory for `NewReader` too? For example

```py
data_directory = os.path.join(settings.SOURCE_DATA_ROOT, corpus.name)
```

or we can add that to `CorpusConfiguration`, so this function can just use

```py
data_directory = corpus.data_directory
```

In the future, we will still need a way to customise the directory for a corpus, but this should work for now.